### PR TITLE
$(sprint).remove() - Closes #218

### LINKED ISSUE #218: Add delegated listener util, or switch from Sprint
We've been manually creating delegated listeners for a bit now:

- https://github.com/sindresorhus/refined-github/blob/master/extension/content.js#L235
- https://github.com/sindresorhus/refined-github/blob/master/extension/content.js#L249
- https://github.com/sindresorhus/refined-github/blob/master/extension/content.js#L269
-https://github.com/sindresorhus/refined-github/blob/master/extension/content.js#L235
- https://github.com/sindresorhus/refined-github/blob/master/extension/copy-file.js#L14
- https://github.com/sindresorhus/refined-github/blob/master/extension/reactions-avatars.js#L45

Rather than doing this manually everytime, it's probably worthwhile for us to abstract this.

I think we have 2 options:

1. Add our own util to handle this
2. Switch to a different DOM lib besides Sprint ([Zepto](http://zeptojs.com/) is a popular one)

Thoughts?

### PR DESCRIPTION
**Pull request by DrewML**
Stuff I did do:
- Changed a lot of stuff to use jQuery, that couldn't be done with Sprint
- Made our usage of the `$varname` convention consistent

Stuff I did _not_ do:
- Update the dimension calculations in `diffheader.js` to use `jQuery` (I assume that's not a deal breaker for us right now, just didn't have the motivation to refactor all that).

Outstanding Questions:
- Do we want to keep using arrow function expressions as event handler callbacks and pluck `event.target`, or do we want to use plain old function expressions + `this`, which is more the "jQuery way"?

Other notes:
- We have _soo_ many features, I'm not confident with the testing I've done so far. Would like 1 or more people to test drive this before we merge + release.

### CODE REVIEW

**Comment by sindresorhus** [comment]
Awesome!

> Update the dimension calculations in diffheader.js to use jQuery (I assume that's not a deal breaker for us right now, just didn't have the motivation to refactor all that).

That's fine. Not worth it.

> Do we want to keep using arrow function expressions as event handler callbacks and pluck event.target, or do we want to use plain old function expressions + this, which is more the "jQuery way"?

Arrow functions.

-

Can you use the official minimal version of jQuery 3?

**Comment by DrewML** [comment]
> Can you use the official minimal version of jQuery 3?

np, done.

File: extension/diffheader.js [hunk 1]
```diff
@@ -109,10 +107,9 @@ window.diffFileHeader = (() => {
 	const setup = () => {
 		$(window).on('scroll.diffheader', () => diffHeaderFilename());
 		const onResize = utils.debounce(() => diffHeaderFilename(true), 200);
-		$(window).on('resize', onResize);
+		$(window).on('resize.diffheader', onResize);
```
Comment (sindresorhus) [inline extension/diffheader.js]: How does `resize.diffheader` work? Can you link to jQuery docs on it? I've never seen this before.

**Comment by sindresorhus** [comment]
Code changes looks good to me, but we should do some manual testing of the affected areas just to be sure.

@paulmolluzzo LGTY?

File: extension/diffheader.js (hunk 1 above)
Comment (DrewML) [inline extension/diffheader.js]: image

It's a namespace. It makes it easier to remove the event handler later, without needing a ref to the handler (like `add/removeEventListener`) or needing to blow away _all_ bound events of that type.

Comment (jamestalmage) [inline extension/diffheader.js]: That idea should totally be ported to Node.

Comment (DrewML) [inline extension/diffheader.js]: [`EventEmitter2`](https://github.com/asyncly/EventEmitter2) has it. Would be sweet to see in core, though

**Comment by paulmolluzzo** [comment]
> We have soo many features, I'm not confident with the testing I've done so far. Would like 1 or more people to test drive this before we merge + release.

Manual tests I see as passing:

- [x] Infinite scroll on dashboard
- [x] Add Reaction Avatars
- [x] Copy on Y
- [x] Tab to add tabs
- [x] Confirm cancel of comments
- [x] Add edit README link
- [x] patch/diff links added
- [x] Diff header
- [x] Show Names

I'm also not seeing any errors in the console as I navigate around or mess with stuff.

Just to be clear, I only tested functionality that seemed affected in this PR, but if there's a lurking bug some other place I haven't seen it yet.

I'm pretty confident we can :shipit: right now, but if you all want to test drive this for another day go for it.

This is excellent @DrewML!

**Comment by sindresorhus** [comment]
Yay! Finally I don't have to check if Sprint supports the jQuery stuff I know. Thanks Andrew :)

**Pull request merged**